internal/app/grpcserver/impl.go

```diff
@@ -303,6 +303,18 @@ func parseUnaryResponseDefinition(
 	if def != nil {
 		switch respType := def.Response.(type) {
 		case *v1.UnaryResponseDefinition_Error:
+			// If error details were not provided to be returned, then the server
+			// should set the conformance payload as the error details for unary responses
+			if respType.Error.Details == nil {
```

Comment:
It's often an anti-pattern to compare slices to `nil`. Rarely do you really intend to distinguish between nil and an empty slice. So it's safest to use `len(respType.Error.Details) == 0`.

Also, why do this only when error details is nil? I think we want to add this to the details, even if non-nil.